src/libworld/openflights.hpp

```diff
@@ -0,0 +1,420 @@
+        if (v.size() == 0)
+            return true;
+
+        return hasAny<string>(v, [s](string test) {
+            return (iequals(s, test));
+            });
+    }
+};
+
+class OpenFlightDataReader
+{
+private:
+    shared_ptr<HostServices> m_host;
+    shared_ptr<OpenFlightsRoutes> m_datas;
+
+public:
+private:
+public:
+    explicit OpenFlightDataReader(shared_ptr<HostServices> _host) : m_host(_host)
+    {
+        m_datas = make_shared<OpenFlightsRoutes>(m_host);
+    };
+
+public:
+    shared_ptr<WorldRoutes> getWorldRoutes() { return m_datas; }
+
+    // Parsing OpenFlights airpors.dat file.
+    // Generates a mapping from iata to icao
+    //
+    // Columns are :
+    //
+    // - Airport ID Unique OpenFlights identifier for this airport.
+    // - Name       Name of airport. May or may not contain the City name.
+    // - City       Main city served by airport. May be spelled differently from Name.
+    // - Country    Country or territory where airport is located. See Countries to cross-reference to ISO 3166-1 codes.
+    // - IATA       3-letter IATA code. Null if not assigned/unknown.
+    // - ICAO       4-letter ICAO code. Null if not assigned.
+    // - Latitude   Decimal degrees, usually to six significant digits. Negative is South, positive is North.
+    // - Longitude  Decimal degrees, usually to six significant digits. Negative is West, positive is East.
+    // - Altitude   In feet.
+    // - Timezone   Hours offset from UTC. Fractional hours are expressed as decimals, eg. India is 5.5.
+    // - DST        Daylight savings time. One of E (Europe), A (US/Canada), S (South America), O (Australia), Z (New Zealand), N (None) or U (Unknown). See also: Help: Time
+    // - Tz database time zone Timezone in "tz" (Olson) format, eg. "America/Los_Angeles".
+    // - Type       Type of the airport. Value "airport" for air terminals, "station" for train stations, "port" for ferry terminals and "unknown" if not known. In airports.csv, only type=airport is included.
+    // - Source     Source of this data. "OurAirports" for data sourced from OurAirports, "Legacy" for old data not matched to OurAirports (mostly DAFIF), "User" for unverified user contributions. In airports.csv, only source=OurAirports is included.
+    void readAirports(istream &input)
+    {
+        CsvParser::parse(input,
+                         [this](CSVRow &row) {
+                             if ((row.size() == 14) && isValidString(row[4]) && isValidString(row[5]))
+                             {
+                                 string iata(row[4].substr(1, row[4].size() - 2));
+                                 string icao(row[5].substr(1, row[5].size() - 2));
+                                 m_datas->m_airportIata2Icao[iata] = icao;
+                             }
+                         });
+        m_host->writeLog("OPENFLIGHTS| Airports parsed : %d association IATA->ICAO found", m_datas->m_airportIata2Icao.size());
```

Comment:
In the log "OPENFLIGHTS" would be a "component name" (a totally undocumented rule, I admit)
Let's keep all component names aligned to 6 characters, so it will look like a column in the log file.
What about "OPNFLT"?

Reply:
OK, let's go back to code 😃.